### PR TITLE
Add `regexp/no-empty-character-class` rule

### PR DESCRIPTION
**Pull request by ota-meshi**
Related to #325

### CODE REVIEW

**Comment by RunDevelopment** [comment]
I was wondering whether it might be useful to [evaluate](https://rundevelopment.github.io/regexp-ast-analysis/docs/latest/modules.html#matchesnocharacters) the character class? 

This does depart from what ESLint does but it might be useful I think. I imagine this rule as an opposite to `match-any`.

**Comment by ota-meshi** [comment]
That's good idea. Do you think we need to change the rule name?

**Comment by RunDevelopment** [comment]
You mean into e.g. `match-none`? I don't think so. While the rule does deviate from ESLint's implementation a little, the intent is still the same. We just check more thoroughly.

**Comment by ota-meshi** [comment]
Thank you for your opinion. I have updated this rule.

File: lib/rules/no-empty-character-class.ts [hunk 1]
```diff
@@ -1,19 +1,23 @@
...
             // TODO Switch to recommended in the major version.
             // recommended: true,
             recommended: false,
         },
         schema: [],
         messages: {
-            unexpected: "Unexpected empty character class.",
+            empty:
+                "Since this character class is empty, it cannot match all characters.",
+            cannotMatchAll: "This character class cannot match all characters.",
```
Comment (RunDevelopment) [inline lib/rules/no-empty-character-class.ts:20]:
```suggestion
            cannotMatchAny: "This character class cannot match any characters.",
```

File: lib/rules/no-empty-character-class.ts [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
 import type { RegExpVisitor } from "regexpp/visitor"
 import type { RegExpContext } from "../utils"
 import { createRule, defineRegexpVisitor } from "../utils"
 
 export default createRule("no-empty-character-class", {
     meta: {
         docs: {
-            description: "disallow empty character classes",
+            description:
+                "disallow character classes that does not match all characters",
```
Comment (RunDevelopment) [inline lib/rules/no-empty-character-class.ts:10]:
```suggestion
                "disallow character classes that match no characters",
```

File: docs/rules/no-empty-character-class.md [hunk 3]
```diff
@@ -2,19 +2,21 @@
...
 
-> disallow empty character classes
+> disallow character classes that does not match all characters
 
 - :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
 
 ## :book: Rule Details
 
-This rule disallows empty character classes.
+This rule reports character classes that cannot match all characters.
```
Comment (RunDevelopment) [inline docs/rules/no-empty-character-class.md:15]:
```suggestion
This rule reports character classes that cannot match any characters.
```

File: docs/rules/no-empty-character-class.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+> disallow character classes that does not match all characters
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> ***This rule has not been released yet.*** </badge>
+
+## :book: Rule Details
+
+This rule reports character classes that cannot match all characters.
+
+The character class that cannot match all characters is an empty character class and a negative character class that contains all characters.
```
Comment (RunDevelopment) [inline docs/rules/no-empty-character-class.md:17]:
```suggestion
Character classes that cannot match any characters are either empty or negated character classes of elements that contain all characters.
```

File: lib/rules/no-empty-character-class.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            category: "Possible Errors",
+            // TODO Switch to recommended in the major version.
+            // recommended: true,
+            recommended: false,
+        },
+        schema: [],
+        messages: {
+            empty:
+                "Since this character class is empty, it cannot match all characters.",
```
Comment (RunDevelopment) [inline lib/rules/no-empty-character-class.ts:19]:
```suggestion
                "This character class matches no characters because it is empty.",
```

**Review by RunDevelopment — CHANGES_REQUESTED** [review]
There was some confusion around "any" and "all".

**Comment by ota-meshi** [comment]
Thank you for the review! I have changed this PR.

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**